### PR TITLE
Allow to get UIWindowType enum value from type

### PR DESCRIPTION
**Pull request by AsesinoBlade**
I was looking for a method to return the type of window that is currently up.   I created GetType in UIWindowFactory to achieve this functionality.

An example of use would be 

            var manager = (UserInterfaceManager)sender;
            if (UIWindowFactory.GetType(manager.TopWindow.GetType()) == UIWindowType.Trade) {
…

thanks 
a

### CODE REVIEW

File: Assets/Scripts/Game/UserInterfaceWindows/UIWindowFactory.cs [hunk 1]
```diff
@@ -132,6 +134,16 @@ public static IUserInterfaceWindow GetInstanceWithArgs(UIWindowType windowType,
             return GetInstance(windowType, args);
         }
 
+        public static UIWindowType GetType(Type windowType)
+        {
+            foreach (var pair in uiWindowImplementations)
+            {
+                if (pair.Value == windowType)
+                    return pair.Key;
```
Comment (TheLacus) [inline Assets/Scripts/Game/UserInterfaceWindows/UIWindowFactory.cs]: That's fine, but you can also use Linq for one line: 

```cs
var pair = uiWindowImplementations.FirstOrDefault(x => x. Value == windowType);
```

**Comment by AsesinoBlade** [comment]
Thanks TheLacus, I made the changes you requested.  I hadn't used Linq before; it is very convenient :)

thanks
a

**Comment by TheLacus** [comment]
Added check for default value (which is null for `Type` class). Sorry i wasn't completly clear, `FirstOrDefault` refers to `KeyValuePair` which is a struct so is never null. It's the value that is null if not found :)

**Comment by AsesinoBlade** [comment]
> Added check for default value (which is null for `Type` class). Sorry i wasn't completely clear, `FirstOrDefault` refers to `KeyValuePair` which is a struct so is never null. The default value is `(default(UIWindowType), null)` so it's the value that is null if not found :)

Ah, thank you for explaining that.  I tested it by passing HUD into it and comparing against Trade.  It worked as expected so I assumed that it returned Null, but now I see why it passed.

I am really enjoying helping out; I am learning so much :)

Thanks again TheLacus!
A

**Pull request merged**